tests/processing/test_functions.py

```diff
@@ -475,3 +480,53 @@ def test_optimize_with_rsa_encryption(tmpdir):
     #     encryption=rsa,
     #     mode="overwrite",
     # )
+
+
+def tokenize(filename: str):
+    with open(filename, encoding="utf-8") as file:
+        text = file.read()
+    text = text.strip().split(" ")
+    word_to_int = {word: random.randint(1, 1000) for word in set(text)}  # noqa: S311
+    tokenized = [word_to_int[word] for word in text]
+
+    yield tokenized
+
+
+@pytest.mark.skipif(sys.platform == "win32", reason="Not tested on windows")
+def test_optimize_race_condition(tmpdir):
```

Comment:
Okay, so I tested it on `AWS t2.2xlarge` for faster speed, and ran @rasbt script in a loop for 100 times (if a single race condition/ error occurs). It didn't.

Then, I copy pasted the code in this test function `test_optimize_race_condition` to see if it fails on github action. But it passes in mac and linux both.

https://github.com/Lightning-AI/litdata/actions/runs/10910350982/job/30280603433?pr=375#step:5:72

I'm unaware of `hyperplane1 machine`.

Can you please clear your cache (in case an already existing bin file exists), and try rerunning the script!
Also, what is your os?